BarrysPeanutsAkkaFlat/src/main/java/barryspeanuts/actor/PaymentActor.java

```diff
@@ -0,0 +1,153 @@
+package barryspeanuts.actor;
+
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+import barryspeanuts.model.CreditCard;
+import barryspeanuts.model.Customer;
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.UUID;
+
+public class PaymentActor extends AbstractBehavior<Object> {
+  private PaymentActor(ActorContext<Object> context) {
+    super(context);
+  }
+
+  public static Behavior<Object> create() {
+    return Behaviors.setup(PaymentActor::new);
+  }
+
+  public static Behavior<Object> behavior() {
+    return Behaviors.setup(PaymentActor::new);
+  }
+
+  @Override
+  public Receive<Object> createReceive() {
+    return newReceiveBuilder().onMessage(PaymentInfo.class, this::handlePayment).build();
+  }
+
+  private Behavior<Object> handlePayment(PaymentInfo msg) {
+    CreditCard creditCard = msg.getCreditCard();
+
+    // Get purchase state
+    getContext()
+        .getLog()
+        .info(
+            "TODO: Getting purchase state for payment according to PaymentId {} or "
+                + "relying upon the Akka Event Sourcing framework",
+            msg.purchaseId);
+    BigDecimal amount = msg.getPaymentAmount();
+    // Now pay
+    getContext()
+        .getLog()
+        .info(
+            "Paying with Credit Card for {} with Credit Card Number {} for the amount of {}",
```

Comment:
You are inconsistent with whether you use trailing punctuation

Reply:
addressed